src/Native/Unix/System.Native/pal_process.c

```diff
@@ -386,71 +385,80 @@ static uint64_t ConvertFromNativeRLimitInfinityToManagedIfNecessary(rlim_t value
     if (value == RLIM_INFINITY)
         return UINT64_MAX;
 
-    return UnsignedCast(value);
+    assert(value >= 0);
+    return (uint64_t)value;
 }
 
-static void ConvertFromRLimitManagedToPal(const RLimit& pal, rlimit& native)
+static void ConvertFromRLimitManagedToPal(const RLimit* pal, struct rlimit* native)
 {
-    native.rlim_cur = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal.CurrentLimit);
-    native.rlim_max = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal.MaximumLimit);
+    native->rlim_cur = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal->CurrentLimit);
+    native->rlim_max = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal->MaximumLimit);
 }
 
-static void ConvertFromPalRLimitToManaged(const rlimit& native, RLimit& pal)
+static void ConvertFromPalRLimitToManaged(const struct rlimit* native, RLimit* pal)
 {
-    pal.CurrentLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native.rlim_cur);
-    pal.MaximumLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native.rlim_max);
+    pal->CurrentLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native->rlim_cur);
+    pal->MaximumLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native->rlim_max);
 }
 
-extern "C" int32_t SystemNative_GetRLimit(RLimitResources resourceType, RLimit* limits)
+#if defined __USE_GNU && !defined __cplusplus
+typedef __rlimit_resource_t rlimitResource;
+typedef __priority_which_t priorityWhich;
```

Comment:
Where are these used?

Reply:
They are used to cast on line 418, 554, 556 etc. I had to check the headers in multiple distros. In Alpine, they are defined as int, in OSX and all the rest of supported Linux, it is __rlimit_resource_t etc. This definition is taken from Ubuntu's header (with `#if defined`).

Reply:
In typical linux, this is the definition. https://github.com/caliston/gccsdk/blob/c8743f53b7fb748905bbe383a83702b53442ccb8/gcc4/recipe/files/gcc/libunixlib/include/sys/resource.h#L16.
In Alpine's sys/resource.h, it is not defined at all, so we cannot use `__priority_which_t`. By defining it here with our camel-cased name, we are able to cast.

Reply:
The Linux man page for getrlimit says the first parameter is int. So it seems we should be casting the argument to int.

Reply:
The error on Linux leg is:

```
15:42:09 /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_arm64+TestOuter_false_prtest/src/Native/Unix/System.Native/pal_process.c:410:28: error: implicit conversion changes signedness: 'int32_t' (aka 'int') to '__rlimit_resource_t' (aka 'enum __rlimit_resource') [-Werror,-Wsign-conversion]
```

https://ci3.dot.net/job/dotnet_corefx/job/master/job/linux-TGroup_netcoreapp+CGroup_Release+AGroup_arm64+TestOuter_false_prtest/3260/console

so i am keeping the typedef for now. It took me 4-5 hours to figure out this block, please let me know if there is a better way around the platform disparity.

Reply:
I've looked into it out of interest and found no better way. The only way you can make glibc provide the functions with the signatures where the first arg is int is to not to define the _GNU_SOURCE. However, we need to have it defined to get access to the `ucred` struct. So let's keep it this way.